### ISSUE TITLE
Low Connect Speed and high delay when using WPA3

### ISSUE THREAD

**Issue by ZhiShengYuan** [OpenIntelWireless/itlwm#701]
**TL; DR** https://github.com/OpenIntelWireless/itlwm/issues/701#issuecomment-966372597

---

**Have You Read Our Docs**
yes

**Are You Reporting A Bug**
yes
**Environment**

- Kext Version: 2.0
- WiFi Card Model: ax201
- PCI Product ID: 
- macOS Version: 11.6

**Description**
when i use macOS 11.6 and 2.0.0 stable driver connect to WIFI-6 wifi, the connection speed seems very low(maybe just 10mbps), but it's normal when I connect to wifi-5 only devices(MI R3G)
is any information I can offer?

**Kext Download Source**
official GitHub,URL is `https://github.com/OpenIntelWireless/itlwm/releases/download/v2.0.0/AirportItlwm_v2.0.0_stable_BigSur.kext.zip`

**Comment by Ardentwheel** [comment]
> Any logs?

AX201
Redmi AX6

[wifi.log](https://github.com/OpenIntelWireless/itlwm/files/7491035/wifi.log)

**Comment by wzgong** [comment]
Kext Version: 1.3.0 through 2.1.0 alpha (log produced with 2.1.0 alpha)
Wifi Card Model: ax210
PCI Product ID: 0x02F0 Sub 0x0074
macOS Version: 11.6/12.0.1 (log produced with 12.0.1)

The router is Redmi AX6. Random high latency (ranging from 100ms to 4000ms+) when pinging the router itself. 
The problem occurs when the WiFi works in WPA2/WPA3 mixed mode. Haven't tried WPA3 mode, but the problem seems to resolve itself when working in WPA2 mode.
Switching channels/channel bandwidth/between ax&ac doesn't seem to have an impact.

Redmi AX6 下碰见超高延迟。Ping 路由器会出现 100-4000ms 的延迟。
WPA2/WPA3混合模式下出现问题。加密切至 WPA2 后问题消失（到目前为止），没有测试 WPA3 模式。
切换频道、频道带宽或者切换ac/ax问题依旧存在。

[itlwm_logs.log](https://github.com/OpenIntelWireless/itlwm/files/7502533/itlwm_logs.log)
Log produced by HeliPort. 上为 HeliPort 导出的日志。

**Comment by zxystd** [comment]
@Ardentwheel Log is incomplete, please follow the guide to collect again. https://openintelwireless.github.io/itlwm/Troubleshooting.html#runtime-logs

**Comment by zxystd** [comment]
> The router is Redmi AX6. Random high latency (ranging from 100ms to 4000ms+) when pinging the router itself. The problem occurs when the WiFi works in WPA2/WPA3 mixed mode. Haven't tried WPA3 mode, but the problem seems to resolve itself when working in WPA2 mode. Switching channels/channel bandwidth/between ax&ac doesn't seem to have an impact.
> 

@wzgong  Do not use WP3 only or WPA2/WPA3 mixed mode WiFi, WPA3 is not supported.

**Comment by Ardentwheel** [comment]
> > The router is Redmi AX6. Random high latency (ranging from 100ms to 4000ms+) when pinging the router itself. The problem occurs when the WiFi works in WPA2/WPA3 mixed mode. Haven't tried WPA3 mode, but the problem seems to resolve itself when working in WPA2 mode. Switching channels/channel bandwidth/between ax&ac doesn't seem to have an impact.
> 
> @wzgong Do not use WP3 only or WPA2/WPA3 mixed mode WiFi, WPA3 is not supported.

Solved, the reason for high latency is that WPA3 is not supported. 
Set to WPA2 solved the problem.

![截屏2021-11-09 下午5 17 48](https://user-images.githubusercontent.com/8460881/140896669-6a9f92d2-a75d-4e6b-b427-28055cf6dcec.png)

**Comment by JJTech0130** [comment]
I have this problem with my AX201, thanks for the info. Unfortunately, my network is WPA3 and I cannot change this, is there any workaround, or way to force the device to use WPA2?

**Comment by wzgong** [comment]
> I have this problem with my AX201, thanks for the info. Unfortunately, my network is WPA3 and I cannot change this, is there any workaround, or way to force the device to use WPA2?

My experience is that it worked fine with version 1.2.0 on a WPA3/WPA2 mixed mode network.

**Comment by JJTech0130** [comment]
As opposed to version 2.0? I’m getting unusable speeds (~5 MB/s)… I’ve talked to the network administrator, and it may be possible to revert to WPA2 only for now, but this issue is annoying.

**Comment by zxystd** [comment]
@JJTech0130 @ZhiShengYuan Temporary solution is using itlwm.kext instead. I will try to implement `11w` feature on gen2/gen3 drivers soon, then the performance would be fine on wpa2/wpa3 network.

**Comment by ZhiShengYuan** [comment]
> @JJTech0130 @ZhiShengYuan Temporary solution is using `itlwm.kext` instead. I will try to implement `11w` feature on gen2/gen3 drivers soon, then the performance would be fine on wpa2/wpa3 network.

well, thanks for your help, I closed the wap3 support in the router at last

**Comment by ZhiShengYuan** [comment]
> I have this problem with my AX201, thanks for the info. Unfortunately, my network is WPA3 and I cannot change this, is there any workaround, or way to force the device to use WPA2?

**Comment by williambj1** [comment]
We'll keep this open

**Comment by JJTech0130** [comment]
Thanks! Unfortunately, I won't be able to test it as I convinced my network administrator to switch back to WPA2... sorry 😅

**Comment by wzgong** [comment]
> [Big Sur.zip](https://github.com/OpenIntelWireless/itlwm/files/7539606/Big.Sur.zip) [Monterey.zip](https://github.com/OpenIntelWireless/itlwm/files/7539609/Monterey.zip) [Catalina.zip](https://github.com/OpenIntelWireless/itlwm/files/7539625/Catalina.zip)
> 
> @ZhiShengYuan @JJTech0130 @wzgong @Ardentwheel Hi, now we are able to connect to wpa2/wpa3 mixed mode AP, please test it.

Working as expected on Monterey. No issues detected so far.

**Comment by zxystd** [comment]
@wzgong Thank you, I also noticed this problem, so it is not merge now, still finding how to resolve.

**Comment by Ardentwheel** [comment]
> [Big Sur.zip](https://github.com/OpenIntelWireless/itlwm/files/7539606/Big.Sur.zip) [Monterey.zip](https://github.com/OpenIntelWireless/itlwm/files/7539609/Monterey.zip) [Catalina.zip](https://github.com/OpenIntelWireless/itlwm/files/7539625/Catalina.zip)
> 
> @ZhiShengYuan @JJTech0130 @wzgong @Ardentwheel Hi, now we are able to connect to wpa2/wpa3 mixed-mode AP, please test it.

ping is unstable
![截屏2021-11-29 上午12 57 10](https://user-images.githubusercontent.com/8460881/143778132-ffb66174-f543-468b-ba7d-9593236b3d28.png)

**Comment by zxystd** [comment]
I think the latest stable release have resolve this issue, please try that:)

**Comment by gzyhao** [comment]
> I think the latest stable release have resolve this issue, please try that:)

Working well. No issue waking from sleep so far ;-)

**Comment by wzgong** [comment]
> I think the latest stable release have resolve this issue, please try that:)

Working well. No issue waking up from sleep ;-)
